src/SDDK/wave_functions.cpp

```diff
@@ -411,31 +416,15 @@ mdarray<T, 1> Wave_functions<T>::sumsqr(device_t pu__, spin_range spins__, int n
             }
             case device_t::GPU: {
 #if defined(SIRIUS_GPU)
-                if(std::is_same<T, double>::value) {
-                    add_square_sum_gpu_double(
-                        reinterpret_cast<std::complex<double> const*>(pw_coeffs(is).prime().at(memory_t::device)),
-                        pw_coeffs(is).num_rows_loc(), n__, gkvecp_.gvec().reduced(), comm_.rank(),
+                add_square_sum_gpu(
+                    reinterpret_cast<std::complex<T> const*>(pw_coeffs(is).prime().at(memory_t::device)),
+                    pw_coeffs(is).num_rows_loc(), n__, gkvecp_.gvec().reduced(), comm_.rank(),
+                    reinterpret_cast<double*>(s.at(memory_t::device)));
```

Comment:
```suggestion
                    reinterpret_cast<T*>(s.at(memory_t::device)));
```
The same at the next call.
Maybe casting is not even required, depending on what types `pw_coeffs` and `s` hold.

Reply:
Yeah, that's much better.